File: FastMines_core/src/fmg/core/img/RotateTransformer.java
package fmg.core.img;

/** Transforming of rotate angle */
public class RotateTransformer implements IModelTransformer {

    @Override
    public void execute(IAnimatedModel model) {
        if (!(model instanceof AnimatedImageModel))
            throw new IllegalArgumentException("Illegal usage transformer");

        AnimatedImageModel am = (AnimatedImageModel)model;

        double rotateAngleDelta = 360.0 / am.getTotalFrames(); // 360° / TotalFrames
        if (!am.getAnimeDirection())
            rotateAngleDelta = -rotateAngleDelta;

      //am.setRotateAngle(am.getCurrentFrame() * rotateAngleDelta); // не враховує початкове значення кута
        am.setRotateAngle(am.getRotateAngle() + rotateAngleDelta);
    }

}

File: FastMines_core/src/fmg/core/img/RotateTransformer.cs
using System;

namespace Fmg.Core.Img {

    /// <summary> Transforming of rotate angle </summary>
    public class RotateTransformer : IModelTransformer {

        public void Execute(IAnimatedModel model) {
            if (!(model is AnimatedImageModel am))
                throw new InvalidOperationException("Illegal usage transformer");

            double rotateAngleDelta = 360.0 / am.TotalFrames; // 360° / TotalFrames
            if (!am.AnimeDirection)
                rotateAngleDelta = -rotateAngleDelta;

          //am.RotateAngle = am.CurrentFrame * rotateAngleDelta; // не враховує початкове значення кута
            am.RotateAngle += rotateAngleDelta;
        }

    }

}
